.github/workflows/ruby-core.yml

```diff
@@ -0,0 +1,53 @@
+name: Ruby Core
+
+concurrency:
+  group: "${{github.workflow}}-${{github.ref}}"
+  cancel-in-progress: true
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  test-all:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Set up latest ruby head
+        uses: ruby/setup-ruby@250fcd6a742febb1123a77a841497ccaa8b9e939 # v1.152.0
+        with:
+          ruby-version: head
+          bundler: none
+      - name: Save latest buildable revision to environment
+        run: echo "REF=$(ruby -e 'print RUBY_REVISION')" >> $GITHUB_ENV
+      - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v3.1.0
+        with:
+          repository: ruby/ruby
+          path: ruby/ruby
+          fetch-depth: 1
+      - name: Checkout the latest buildable revision
+        run: |
+          git fetch --depth=1 origin ${{ env.REF }}
+          git switch -c ${{ env.REF }}
```

Comment:
What does this do? Is it necessary? Isn't it enough to pick latest ruby/ruby master?